cyten/models/couplings.py

```diff
@@ -353,7 +352,7 @@ def onsite_interaction(sites: list[BosonicDOF] | list[FermionicDOF], U: float =
     return Coupling.from_dense_block(h, sites=sites, name=name, understood_braiding=True)
 
 
-def density_density_interaction(sites: list[BosonicDOF] | list[FermionicDOF], V: float = 1,
+def density_density_interaction(sites: list[OccupationDOF], V: float = 1,
```

Comment:
that type hint was intentional.
It should be all bosons or all fermions.
The new type hint would allow a mixed list

Reply:
We would already get an error from mismatching symmetry when the coupling tries to build a SymmetricTensor, but it is probably a good idea to error early with a useful error message